platypus-core/src/main/java/com/platypus/crw/udp/UdpConstants.java

```diff
@@ -162,7 +162,13 @@ public static void writeSensorData(DataOutputStream out, SensorData sensor) thro
     public static SensorData readSensorData(DataInputStream in) throws IOException {
         SensorData sensor = new SensorData();
         sensor.channel = in.readInt();
-        sensor.type = SensorType.values()[Math.max(in.readByte(), SensorType.values().length - 1)];
+        byte type_index = in.readByte();
+        if (type_index < 0) {
+            sensor.type = SensorType.UNKNOWN;
+        }
+        else {
+            sensor.type = SensorType.values()[Math.min(type_index, SensorType.values().length - 1)];
+        }
```

Comment:
Good catch on this bug!

Instead of branching, can we just do:
```
int type_index = in.readUnsignedByte();
sensor.type = SensorType.values()[Math.min(type_index, SensorType.values().length - 1)];
```

I think that produces the same logic without having a branch for negative numbers.